src/App.tsx

```diff
@@ -116,75 +126,97 @@ const hexRegex = /^#[a-f0-9]{6}$/gi;
 
 // TODO: add an error boundary for parse fails
 const Editor: Component<{ currentFile: FileSystemFileHandle }> = (props) => {
-  const [contents, { mutate }] = createResource(async () => {
+  const [obj, setObject] = createStore<Record<string, unknown>>({});
```

Comment:
random idea to make this type slightly narrower? I know `JSON.parse` gives us `any`, but we know it's really the following type:

```ts
type Json = string | number | boolean | null | { [key: string]: Json } | Json[];
```

so we could do this:

```suggestion
  const [obj, setObject] = createStore<Record<string, Json>>({});
```

Reply:
great idea! thinking more along these lines, i guess json doesn't have to be an object too. maybe this editor should support top level primitive values and array json too.